src/set_theory/game/basic.lean

```diff
@@ -218,6 +218,26 @@ rfl
    = x.move_right i * y + x * y.move_left j - x.move_right i * y.move_left j :=
 by { cases x, cases y, refl }
 
+lemma left_moves_mul_cases {x y : pgame} (k : (x * y).left_moves) :
+  (∃ ix iy, k = to_left_moves_mul (sum.inl ⟨ix, iy⟩)) ∨
+  ∃ jx jy, k = to_left_moves_mul (sum.inr ⟨jx, jy⟩) :=
+begin
+  rw ←to_left_moves_mul.apply_symm_apply k,
+  rcases to_left_moves_mul.symm k with ⟨ix, iy⟩ | ⟨jx, jy⟩,
+  { exact or.inl ⟨ix, iy, rfl⟩ },
+  { exact or.inr ⟨jx, jy, rfl⟩ }
+end
```

Comment:
This feels like it would be more convenient written as an induction principle on `k`

Reply:
```suggestion
lemma left_moves_mul_cases {x y : pgame} (P : (x * y).left_moves -> Prop) :
  (\forall ix iy, P $ to_left_moves_mul (sum.inl ⟨ix, iy⟩))
  (\forall jx jy, P $ to_left_moves_mul (sum.inr ⟨jx, jy⟩)) (k) : P k :=
sorry
```

Reply:
This seems like an interesting idea. I'll try this out on both the add and mul lemmas and PR if it simplifies proofs.